mix.exs

```diff
@@ -29,6 +29,7 @@ defmodule Opencensus.Absinthe.MixProject do
 
   def application() do
     [
+      applications: [:opencensus, :absinthe],
```

Comment:
You do not need to specify these, it should be handled by Mix OOtB.

Reply:
From my reading of the docs ([Mix.Tasks.Compile.App](https://hexdocs.pm/mix/Mix.Tasks.Compile.App.html) and [Mix.Tasks.Deps](https://hexdocs.pm/mix/Mix.Tasks.Deps.html#module-dependency-definition-options)), I need to either:

* Scatter `runtime: false` through the `deps/0` result, which @seancribbs insisted I shouldn't do, and

* Specify `applications` in the `application/0` result, which you insist I shouldn't do.

Reviewing all the `mix.exs` files in my work app's dependency graph, I think Sean's advice might have a gap: if `only` is absent or if `only` includes `dev` or `prod`, maybe you _do_ need `runtime: false`. Citations: 

* `argon2_elixir` 
* `artificery` 
* `combine` 
* `comeonin` 
* `credo` 
* `distillery` (including for `ex_unit_clustered_case` in `test`, which might be a mistake), 
* `erlex` 
* `jason` 
* `opencensus_elixir` 
* `opencensus_plug` 
* `sentry`
    
I left out the three I can commit to all by myself.

That said, I also found plenty of use of `applications`:

* `absinthe_plug`
* `credo`
* `db_connection`
* `bamboo`
* `jose`
* `phoenix_pubsub`
* `mix_test_watch`
* `sentry`
* `postgrex`
* `sweet_xml`
* `timex`
* `tzdata`
* `tesla`, which has machinery for tweaking it based on `Mix.env/0`

I left out the ones only specifying `logger`, which might also be a mistake.

If you've got any specific material that'll help straighten this out, I'm all ears.

Reply:
I'd mis-remembered Sean's advice: he said compile-time-only, which I forgot, and non-BEAM rather than non-Elixir. So, the list of `runtime: false` should be longer because I erroneously excluded all the Erlang. That said, what's left is benchmarking, types checking, build, parameter based testing, and document generation… it's all compile-time. And the `test` ones are plausibly about compiling the tests.

Zooming back out, all I'm trying to do is to _not_ start `absinthe_plug` if it's absent.

Reply:
I've consolidated to the majority opinion across my work app's dependencies. It doesn't have `applications`, there's only one `runtime: false` left, and it'll do for now.